examples/vision/deit.py

```diff
@@ -0,0 +1,576 @@
+        self.distillation_loss_fn = distillation_loss_fn
+
+    def train_step(self, data):
+        # Unpack data.
+        x, y = data
+
+        # Forward pass of teacher
+        teacher_predictions = tf.nn.softmax(self.teacher(x, training=False), -1)
+        teacher_predictions = tf.argmax(teacher_predictions, -1)
+
+        with tf.GradientTape() as tape:
+            # Forward pass of student.
+            cls_predictions, dist_predictions = self.student(x / 255.0, training=True)
+
+            # Compute losses.
+            student_loss = self.student_loss_fn(y, cls_predictions)
+            distillation_loss = self.distillation_loss_fn(
+                teacher_predictions, dist_predictions
+            )
+            loss = (student_loss + distillation_loss) / 2
+
+        # Compute gradients.
+        trainable_vars = self.student.trainable_variables
+        gradients = tape.gradient(loss, trainable_vars)
+
+        # Update weights.
+        self.optimizer.apply_gradients(zip(gradients, trainable_vars))
+
+        # Update the metrics configured in `compile()`.
+        student_predictions = (cls_predictions + dist_predictions) / 2
+        self.compiled_metrics.update_state(y, student_predictions)
+
+        # Return a dict of performance.
+        results = {m.name: m.result() for m in self.metrics}
+        results.update(
+            {"student_loss": student_loss, "distillation_loss": distillation_loss}
+        )
+        return results
+
+    def test_step(self, data):
+        # Unpack the data.
+        x, y = data
+
+        # Compute predictions.
+        y_prediction = self.student(x / 255.0, training=False)
+
+        # Calculate the loss.
+        student_loss = self.student_loss_fn(y, y_prediction)
+
+        # Update the metrics.
+        self.compiled_metrics.update_state(y, y_prediction)
+
+        # Return a dict of performance.
+        results = {m.name: m.result() for m in self.metrics}
+        results.update({"student_loss": student_loss})
+        return results
+
+    def call(self, inputs):
+        return self.student(inputs / 255.0, training=False)
+
+
+"""
+## Load the teacher model
+
+This model is based on the BiT family of ResNets
+([Kolesnikov et al.](https://arxiv.org/abs/1912.11370))
+fine-tuned on the `tf_flowers` dataset. You can refer to
+[this notebook](https://github.com/sayakpaul/deit-tf/blob/main/notebooks/bit-teacher.ipynb)
+to know how the training was performed. The teacher model has about 212 Million parameters
+which is about **40x more** than the student.
+"""
+
+"""shell
+wget -q
+https://github.com/sayakpaul/deit-tf/releases/download/v0.1.0/bit_teacher_flowers.zip 
+unzip -q bit_teacher_flowers.zip 
+"""
+
+# teacher_model_file = keras.utils.get_file( 
+#     origin="https://github.com/sayakpaul/deit-tf/releases/download/v0.1.0/bit_teacher_flowers.zip",
+#     extract=True,
+#     cache_subdir="models"
+# )
+bit_teacher_flowers = keras.models.load_model("bit_teacher_flowers")
```

Comment:
I intentionally kept this block. 

If I use it to download the weights I run into:

```
FileExistsError: [Errno 17] File exists: '/root/.keras/datasets/bit_teacher_flowers'.
```